modules/ingest-otel/src/main/java/org/elasticsearch/ingest/otel/NormalizeForStreamProcessor.java

```diff
@@ -103,6 +108,41 @@ public IngestDocument execute(IngestDocument document) {
 
         // non-OTel document
 
+        // handling structured messages
+        Map<String, Object> body = null;
+        try {
+            String message = document.getFieldValue("message", String.class, true);
```

Comment:
I'm not sure what `document.getFieldValue("message", String.class, true);` does exactly, but I guess we don't want to auto-cast anything here - the rest of the logic should only apply when message is a string.

If `getFieldValue` throws if message isn't a string I think we are good here.